doc/source/whatsnew/v0.24.0.txt

```diff
@@ -235,7 +235,7 @@ Other API Changes
   a ``KeyError`` (:issue:`21678`).
 - Invalid construction of ``IntervalDtype`` will now always raise a ``TypeError`` rather than a ``ValueError`` if the subdtype is invalid (:issue:`21185`)
 - Trying to reindex a ``DataFrame`` with a non unique ``MultiIndex`` now raises a ``ValueError`` instead of an ``Exception`` (:issue:`21770`)
--
+- :class:`Float64HashTable` handles zeros/signed zeros and all flavors of NaNs consistently: it is no longer possible to have both, zero and signed-zero, as keys at the same time in a table, also there can be at most one NaN-key in a table (:issue:`21866`)
```

Comment:
this is not public, my question below was whether this has a public change?

Reply:
It fixes the bug #21866, i.e. produces right results for some esoteric corner cases. This change of behavior can be observed by the end-user, but is this then a public change worth mentioning?

Reply:
https://github.com/pandas-dev/pandas/issues/21866#issuecomment-404402552

is this the public case? you can list this, but just make it related to the effect on ``.unique()``

Reply:
also add a test w.r.t. unique